pkg/operator/target_config_reconciler_v311_00.go

```diff
@@ -106,6 +127,33 @@ func manageKubeApiserverConfigMap_v311_00_to_latest(client coreclientv1.ConfigMa
 	return resourceapply.ApplyConfigMap(client, requiredConfigMap)
 }
 
+func manageSessionSecret_v311_00_to_latest(client coreclientv1.SecretsGetter) (*corev1.Secret, bool, error) {
```

Comment:
this approach doesn't look right.  We know this is likely to move and we know that it needs to rotate.  Make a different control loop that manages creation and rotation by itself.  Then you simply list the name to be copied down.

Reply:
Ah, ok. I'll revise this.